src/julia-parser.scm

```diff
@@ -865,7 +865,7 @@
            (begin
              #;(if (and (number? ex) (= ex 0))
                  (error "juxtaposition with literal \"0\""))
-             `(call * ,ex ,(parse-unary s))))
+             `(call .* ,ex ,(parse-unary s))))
```

Comment:
I'm not convinced this revised behaviour is a good thing. This could break things like `2x` where `x::Diagonal`, which `broadcast` will try to promote to `Array`.